cmd/antrea-agent/agent.go

```diff
@@ -244,7 +244,9 @@ func run(o *Options) error {
 		multicastEnabled,
 		o.config.SNATFullyRandomPorts,
 		*o.config.Egress.SNATFullyRandomPorts,
-		serviceCIDRProvider)
+		serviceCIDRProvider,
+		wireguardConfig.Port,
```

Comment:
Another concern is about the WireGuard port check, looks like we didn't check the range before:
```
	if o.config.WireGuard.Port == 0 {
		o.config.WireGuard.Port = apis.WireGuardListenPort
	}
```
I suppose we should go through and update the validation to make sure it's a valid port? @antoninbas

Reply:
That sounds good to me but we should do it in a separate PR, we have other ports that we could validate as well, e.g. the `tunnelPort`. Can you open an issue to track this?

Reply:
Issue created: https://github.com/antrea-io/antrea/issues/7035, I will take a look at it later today.

Reply:
We already have this logic: https://github.com/antrea-io/antrea/blob/main/cmd/antrea-agent/options.go#L439